### PR TITLE
ssh/tailssh: restore support for recording locally

### PR DESCRIPTION
**Pull request by maisem**
Updates tailscale/corp#9967

### CODE REVIEW

File: ssh/tailssh/tailssh.go [hunk 1]
```diff
@@ -1154,6 +1155,10 @@ func (ss *sshSession) run() {
 	return
 }
 
+// recordSSHToLocalDisk is a temporary dev knob to test the SSH recording
+// functionality and support off-node streaming.
```
Comment (twitchyliquid64) [inline ssh/tailssh/tailssh.go:1159]: > and support off-node streaming

This knob is just for enabling local storage, no?

File: ssh/tailssh/tailssh.go [hunk 2]
```diff
@@ -1510,12 +1515,33 @@ func (ss *sshSession) connectToRecorder(ctx context.Context, recs []netip.AddrPo
...
+}
+
 // startNewRecording starts a new SSH session recording.
 // It may return a nil recording if recording is not available.
 func (ss *sshSession) startNewRecording() (_ *recording, err error) {
 	recorders, onFailure := ss.recorders()
+	var localRecording bool
 	if len(recorders) == 0 {
-		return nil, errors.New("no recorders configured")
+		if recordSSHToLocalDisk() {
```
Comment (twitchyliquid64) [inline ssh/tailssh/tailssh.go:1540]: I guess this implies that local recording is only done if remote recording wasnt requested? we should document that somewhere.

File: ssh/tailssh/tailssh.go (hunk 1 above)
Comment (maisem) [reply ssh/tailssh/tailssh.go]: yep, updated docs

File: ssh/tailssh/tailssh.go (hunk 2 above)
Comment (maisem) [reply ssh/tailssh/tailssh.go:1541]: updated docs

**Review by twitchyliquid64 — APPROVED** [review]

**Pull request merged**